basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/ElytronPrincipalDecoderTestCase.java

```diff
@@ -0,0 +1,686 @@
+                        .build());
+        } finally {
+            ops.removeIfExists(concatenatingPrincipalDecoderAddress);
+            ops.removeIfExists(constantPrincipalDecoder1address);
+            ops.removeIfExists(constantPrincipalDecoder2address);
+            ops.removeIfExists(constantPrincipalDecoder3address);
+            adminOps.reloadIfRequired();
+        }
+    }
+
+    /**
+     * @tpTestDetails Create Elytron Concatenating Principal Decoder instance in model
+     * and try to remove it in Web Console's Elytron subsystem configuration.
+     * Validate the resource is not any more visible in Concatenating Principal Decoder table.
+     * Validate created resource is not any more present in the model.
+     */
+    @Test
+    public void removeConcatenatingPrincipalDecoderTest() throws Exception {
+        String
+            concatenatingPrincipalDecoderName = randomAlphanumeric(5),
+            constantPrincipalDecoder1name = randomAlphanumeric(5),
+            constantPrincipalDecoder2name = randomAlphanumeric(5),
+            constant1value = randomAlphanumeric(5),
+            constant2value = randomAlphanumeric(5);
+        Address
+            concatenatingPrincipalDecoderAddress =
+                elyOps.getElytronAddress(CONCATENATING_PRINCIPAL_DECODER, concatenatingPrincipalDecoderName),
+            constantPrincipalDecoder1address =
+                elyOps.getElytronAddress(CONSTANT_PRINCIPAL_DECODER, constantPrincipalDecoder1name),
+            constantPrincipalDecoder2address =
+                elyOps.getElytronAddress(CONSTANT_PRINCIPAL_DECODER, constantPrincipalDecoder2name);
+        ResourceVerifier concatenatingPrincipalDecoderVerifier = new ResourceVerifier(concatenatingPrincipalDecoderAddress, client);
+
+        try {
+            ops.add(constantPrincipalDecoder1address, Values.of(CONSTANT, constant1value)).assertSuccess();
+            ops.add(constantPrincipalDecoder2address, Values.of(CONSTANT, constant2value)).assertSuccess();
+            ops.add(concatenatingPrincipalDecoderAddress, Values.of(PRINCIPAL_DECODERS, new ModelNodeListBuilder()
+                    .addAll(constantPrincipalDecoder1name, constantPrincipalDecoder2name)
+                    .build())).assertSuccess();
+            concatenatingPrincipalDecoderVerifier.verifyExists();
+
+            page.navigateToDecoder().selectResource(CONCATENATING_PRINCIPAL_DECODER_LABEL).getResourceManager()
+                    .removeResource(concatenatingPrincipalDecoderName).confirmAndDismissReloadRequiredMessage().assertClosed();
+            assertFalse("Removed resource should not be present in the table any more!",
+                    page.resourceIsPresentInMainTable(concatenatingPrincipalDecoderName));
+            concatenatingPrincipalDecoderVerifier.verifyDoesNotExist();
+        } finally {
+            ops.removeIfExists(concatenatingPrincipalDecoderAddress);
```

Comment:
Shouldn't `constantPrincipalDecoder1address` and `constantPrincipalDecoder2address` be removed here to?